### DISCUSSION TITLE
Is there a walkthrough?

### DISCUSSION THREAD

**Discussion by alekdavisintel** [AnWeber/vscode-httpyac#286] [General]
I'm looking to migrate our Postman regression test suite to a different platform and really like the idea of Visual Studio Code Rest Client and httpYAC, but since it's a somewhat different paradigm, I'm a bit overwhelmed and am wondering if there is a step-by-step tutorial explaining how to accomplish the common tasks, such as:

- Create an application.
- Define environment (base URL of the REST target being tested, etc).
- Define auth information for the environment (e.g. for OAuth2, URL of the token endpoint, client ID, secret, also how to handle authorization code flow). How do you make sure sensitive settings (client secret) does not get leaked when code is checked in source control.
- Configure proxy settings (if needed).
- Define variables.
- Build the first test.
- Build series of test with the variables flowing from one request to another (e.g. call POST to create object, get ID, use ID in GET to validate object).
- Etc.

I read throught the https://httpyac.github.io/ posts, but am still struggling a bit with the overall picture of how to do things correctly. Is there a resource with some walkthroughs for the typical usage scenario?

**Comment by AnWeber** [comment]
There is currently no dedicated step by step guide. Just start creating a file and test some examples in from [Guide](https://httpyac.github.io/guide/request.html) and see what happens. You could use some tools like [fiddler](https://www.telerik.com/fiddler)/ [mitmproxy](https://mitmproxy.org/) and intercept the request, copy it to a http file and exeute it with httpyac.You could be using [this guide for Intellij Http Client](https://www.jetbrains.com/help/idea/exploring-http-syntax.html). My client is compatible with Intellij Http Client.  But I would advise you to read my guide again, as I support a few more simplifications/ features. 
I understand that it seems too complex at first. But the simple case will simply work. I also wanted to say in the docs that more fatures are provied. In principle, you can overwrite, redefine and expand almost any kind of feature of my client. That's on purpose. I also use it for my own non-public plugins

**Comment by alekdavisintel** [comment]
I know that it will work for a simple case (pretty much everything works for a simple case), but we have a reasonably complex scenario (regression testing microservices with hundreds of test cases, manually, ad-hoc, in CI/CD, with strict security requirements, etc). I wrote a utility script that made it a lot easier for us to implement tests in Postman (https://github.com/alekdavis/PostmanUtilities), but after the recent Postman changes (licensing, move to cloud, etc), we are looking for alternatives. If httpYAC works for our use case, we'll need to rewrite a lot of code, but I would not mind, because it may be better in long term.
